pkg/controller/infrastructure/infraflow/reconcile.go

```diff
@@ -314,7 +314,7 @@ func (fctx *FlowContext) findExistingSubnet() (*subnets.Subnet, error) {
 		return nil, err
 	}
 	if networkID == nil {
-		return nil, fmt.Errorf("network not found")
+		return nil, nil
```

Comment:
This feels weird. You could also do 
```golang
		if err != nil || networkID == nil {
		return nil, err
	}
```
That would be equivalent. But I am not sure if it actually an improvement. I'll leave it up to you to decide.

Reply:
Some context then:
- in this function we are looking to find an existing subnet. Some lines above we are trying to figure out the network (as we need that to find our subnet). If can't find a network, well we won't find a subnet.

That being said, I don't think that we should treat the lack of network as an error, or simply as a testament to the lack of a subnet existing (which is what this PR tries to do)

Reply:
Yes. The weirdness (to me) is the explicit handling of this case. Ideally, something like this might possibly be more concisely (yet still explicitly) handled with some ignoreNotFound()-style handling, though that would require some more restructuring (if at all possible).

The change suggested above will not treat the absent networkID as an error, so I'm not sure if that is what you're getting at. But I guess it shows that it is too implicit.

Reply:
> Yes. The weirdness (to me) is the explicit handling of this case.

The error that prevents deletion in https://github.com/gardener/gardener-extension-provider-openstack/issues/890 is coming from the line 317 changed above (it's an error we return and not an API call error). 

In this case I can't make a an API call to get the subnet for I know not the network ID. We essentially recognize that we can't make an API call that would make sense (not without a networkID) and exit early. 

The only change here is whether this is treated as an error.

Reply:
Looking at the places where the func `findExistingSubnet` is used:
- two times in `delete.go`, where with this PR we make sure to only delete subnet or ensure it's ID if a network is found
- one time in `reconcile.go` (ensureSubnet) where also the task dependency ensures that a network must exist

So for me, returning the error as it was before does make sense since we should never reach this case

Reply:
> So for me, returning the error as it was before does make sense since we should never reach this case

Well, the linked issue we try to fix does contain at least one case where we reached that error 😅. In a partial delete for example, we may have deleted the subnet and network and retry the deletion. At which point we will reach that error.

In the reconcile, you won't reach this part, because the `ensureNetwork` step is guaranteed to have finished successfully before reconciling the subnet

Reply:
We could probably skip the `findExistingSubnet` the second time in the deletion, but still if you can't find a network, you won't find a subnet - so I am not sure if this should really be an error.

Reply:
ah my bad, right, not finding the subnet ID is not treated as an error....
forget my comment above